packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -1231,26 +1224,25 @@ class PubspecDependency extends PubspecLine {
   final String version; // the version string if found, or blank.
   final bool isTransitive; // whether the suffix matched kTransitiveMagicString
   final String sourcePath; // the filename of the pubspec.yaml file, for error messages
-  bool isDevDependency; // Whether this dependency is under the `dev dependencies` section.
+  bool? isDevDependency; // Whether this dependency is under the `dev dependencies` section.
 
   DependencyKind get kind => _kind;
   DependencyKind _kind = DependencyKind.normal;
 
   /// If we're a path or sdk dependency, the path or sdk in question.
-  String get lockTarget => _lockTarget;
-  String _lockTarget;
+  String? _lockTarget;
 
   /// If we were a two-line dependency, the second line (see the inherited [line]
   /// for the first).
-  String get lockLine => _lockLine;
-  String _lockLine;
+  String get lockLine => _lockLine!;
```

Comment:
Is this safe? I see existing null checks.
https://github.com/flutter/flutter/blob/87f5f4685c6239d66455f7253210de29a82182ab/packages/flutter_tools/lib/src/commands/update_packages.dart#L915
https://github.com/flutter/flutter/blob/87f5f4685c6239d66455f7253210de29a82182ab/packages/flutter_tools/lib/src/commands/update_packages.dart#L942

Reply:
good catch, made it nullable